### ISSUE TITLE
Implement local https file url

### ISSUE THREAD

**Issue by luckydonald** [tdlight-team/tdlight-telegram-bot-api#42]
So the original Bot servers lets url uploads be handled by the actual telegram backend.

This means the self hosted version of the API in your network doesn't solve the following use cases:

- Having the files accessible only in the local network
- Being able to increase the size limits (5 MB max size for photos and 20 MB max for other types of content.)

The latter being the reason why I looked into self host the API, as 5 MB for images is often not enough.

**Comment by MarcoBuster** [comment]
You can disable file limits as described by #3. 
I don't think add support to https to the project it's a good idea, use a reverse proxy.

**Comment by luckydonald** [comment]
No, I think you might have misunderstood what I meant.

What I meant was like

```http
GET /bot1234:abc/sendPhoto?chat_id=1234&photo=http://192.168.178.69:4458/some_image.png
```

If `192.168.178.69` is a purely local IP address, which is reachable for the bot-api server but not from outside (Telegram servers) it would fail.
Additionally it is not able to increase those file limits when using urls instead of manually downloading and up uploading those yourself or using existing `file_id`s. Lastly, only `https://` seems supported by that.

Which is a bummer, as I actually wanted to increase the size limits with uploads by url, and the fact that you send the API server a url but some other server in some datacenter attempts to download the file instead can be confusing as well.

**Comment by luckydonald** [comment]
@luckydonald https://t.me/TDLightChat/2698
Telegram's markdown is a joke.

**Comment by RamiMohammedSeid** [comment]
Yes this was what happened to me. but you can try downloading the data as a buffer and send it to the local server, I don't think it'd have that much over-head or latency difference.
